### PR TITLE
SOZO-138 | rewrite planner to output highlevel actions

### PR DESCRIPTION
**Pull request by davidlkang**

### CODE REVIEW

File: hlc/plan/planner.py [hunk 1]
```diff
@@ -48,25 +55,40 @@ def plan(grid, grid_dimension:(int, int), start_pos = (0,0), layer = 0, orientat
     pos = start_pos
     plan =  []
 
-    max_layer = max(grid_dimension)/2
+    max_layer = math.floor(max(grid_dimension)/2)
```
Comment (elli-wolf) [inline hlc/plan/planner.py]:
```suggestion
    max_layer = max(grid_dimension) // 2
```

File: hlc/plan/planner.py [hunk 2]
```diff
@@ -7,11 +9,16 @@
 LEFT = (-1, 0)
 LEFT_FORWARD = (-1, 1)
 
+MOVE_FORWARD = 'move forward'
+TURN_RIGHT = 'turn right 90degree'
```
Comment (alexmorten) [inline hlc/plan/planner.py]:
```suggestion
TURN_RIGHT = 'turn right 90 degrees'
```
maybe?

File: hlc/plan/planner.py [hunk 3]
```diff
@@ -48,25 +55,40 @@ def plan(grid, grid_dimension:(int, int), start_pos = (0,0), layer = 0, orientat
...
+    while layer < max_layer:
 
-        plan.append(orientation)
+        plan.append(MOVE_FORWARD)
         pos = update_pos(pos, orientation)
 
-        if pos == start_pos:
-            plan.append(RIGHT_FORWARD)
-            pos = add_pos_tuple(pos, RIGHT_FORWARD)
+        if pos == add_pos_tuple(start_pos, (1,0)):
```
Comment (elli-wolf) [inline hlc/plan/planner.py]: Maybe you should explain what this if-clausel in a comment - it's not very readable

File: hlc/tests/test_planner.py [hunk 4]
```diff
@@ -1,6 +1,25 @@
...
+from enum import Enum
 import numpy as np
 
+class Orientation(Enum):
+    FORWARD = (0,1)
+    RIGHT = (1, 0)
+    BACKWARD = (0, -1)
+    LEFT = (-1, 0)
+
+def change_orientation(orientation):
```
Comment (alexmorten) [inline hlc/tests/test_planner.py]: This looks like a left turn to me

File: hlc/tests/test_planner.py [hunk 5]
```diff
@@ -21,14 +40,52 @@ def test_empty_map_6x6():
     plan_output = planner.plan(grid, grid_dimension, start_pos)
 
     # execute the actions and set explored pos to True
+    orientation = Orientation.FORWARD
     for action in plan_output:
-        pos = planner.add_pos_tuple(pos, action)
-        grid[pos] = True
+        if action == "turn right 90degree":
```
Comment (alexmorten) [inline hlc/tests/test_planner.py]: this should reference the constant

File: hlc/tests/test_planner.py [hunk 6]
```diff
@@ -113,8 +178,8 @@ def test_dead_end_6x6():
 
     # create the grid as a dictionary
     # blocked - False, unexplored - None, explored - True
-    for x in range(6):
-        for y in range(6):
+    for x in range(7):
+        for y in range(7):
```
Comment (alexmorten) [inline hlc/tests/test_planner.py]: ? `test_dead_end_6x6 `

File: hlc/tests/test_planner.py [hunk 7]
```diff
@@ -131,11 +196,15 @@ def test_dead_end_6x6():
 
     # execute the actions and set explored pos to True
     for action in plan_output:
-        pos = planner.add_pos_tuple(pos, action)
-        grid[pos] = True
+        if action == "turn right 90degree":
```
Comment (alexmorten) [inline hlc/tests/test_planner.py]: Should reference the constant

**Review by alexmorten — CHANGES_REQUESTED** [review]

File: hlc/tests/test_planner.py [hunk 8]
```diff
@@ -21,14 +40,52 @@ def test_empty_map_6x6():
...
+    keys_grid = []
+    start_pos = (0, 0)
+    pos = (0, 0)
+    grid_dimension = (4, 5)
+
+    # create the grid as a dictionary
+    # blocked - False, unexplored - None, explored - True
+    for x in range(5):
+        for y in range(6):
+            keys_grid.append((x, y))
```
Comment (elli-wolf) [inline hlc/tests/test_planner.py]: Should reference `grid_dimensions` and this function also generates a grid 5x6, not 4x5.

File: hlc/tests/test_planner.py (hunk 4 above)
Comment (davidlkang) [reply hlc/tests/test_planner.py]: No. This is a right turn.

File: hlc/plan/planner.py (hunk 3 above)
Comment (katethemate) [inline hlc/plan/planner.py]: There is a space missing before 0 in the tuple.

**Review by alexmorten — APPROVED** [review]

**Pull request merged**